database/rrdcalctemplate.c

```diff
@@ -45,9 +45,7 @@ static int rrdcalctemplate_is_there_label_restriction(RRDCALCTEMPLATE *rt,  RRDH
 }
 
 static inline int rrdcalctemplate_test_additional_restriction(RRDCALCTEMPLATE *rt, RRDSET *st) {
-    if (rt->charts_pattern &&
-        !(simple_pattern_matches(rt->charts_pattern, st->id) ||
-         simple_pattern_matches(rt->charts_pattern, st->name)))
+    if (rt->charts_pattern && !simple_pattern_matches(rt->charts_pattern, st->name))
```

Comment:
I think it will be better to remove the exclamation point in front of parentheses, instead to remove the ID.

Reply:
You mean make it `if (matches(id) OR matches(name)) return 0`?

Reply:
Yes, I was thinking to use:

```
    if (rt->charts_pattern &&
        (simple_pattern_matches(rt->charts_pattern, st->id) ||
        simple_pattern_matches(rt->charts_pattern, st->name)))
```

This way we would keep `name` and `id`.

Reply:
That is opposite to our global pattern - we filter if a pattern doesn't match and you are suggesting to filter if it matches.

`simple_pattern_matches` returns:
 - 0: if it doesn't match or the match is negative (! prefix).
 - 1: only when it matches and the match is positive.

Am i missing something?

Reply:
Sorry, my fault.